delta-api/src/main/java/io/cdap/delta/api/SourceTable.java

```diff
@@ -46,8 +57,26 @@ public String getTable() {
     return table;
   }
 
-  public List<SourceColumn> getColumns() {
-    return columns;
+  /**
+   * @return table columns that should be replicated. If empty, all columns should be replicated.
+   */
+  public Set<SourceColumn> getColumns() {
+    // check for null because this object can be created through deserialization of user provided input
+    return columns == null ? Collections.emptySet() : columns;
```

Comment:
Should it be good to make it safe to return unmodified empty set if columns is null?

Reply:
Collections.emptySet() is unmodifiable, but you're right, it's better to wrap columns to make it unmodifiable.

Was wrapping it in the constructor, but the constructor is not called when this is created through deserialization. Will adjust several of these classes accordingly.